pkg/conversion/queryparams/convert_test.go

```diff
@@ -157,6 +165,14 @@ func TestConvert(t *testing.T) {
 			},
 			expected: url.Values{"ptr": {"5"}},
 		},
+		{
+			input: &foo{
+				TestEnv: []testEnv{
+					{Name: "foo", Value: "bar"},
+				},
+			},
+			expected: url.Values{"env.0.name": {"foo"}, "env.0.value": {"bar"}, "str": {""}},
```

Comment:
Since url.Values is actually a list for every key, I guess I would expect lists instead of the index appearing in the key name. Hm, maybe this is better, though. Did you consider doing it that way?